### PR TITLE
Add docs for user linking REST API

### PR DESCRIPTION
**Pull request by chloepouprom**
@jugarrit @jpjoyal @alavers @mspensieri

### CODE REVIEW

File: source/rest/index.md [hunk 1]
```diff
@@ -963,6 +963,128 @@ Suppose for example you begin a conversation with an end user `bob@example.com`
...
+```
+
+<api>`POST /v1/appusers/{smoochId|userId}/channels`</api>
+
+| **Arguments**                 |                            |
+|-------------------------------|----------------------------|
+| **type**<br/><span class='req'>required</span>     | The channel to link.|
+| **{entity}**<br/><span class='req'>required</span> | The required entity for linking. This is [different for each channel](#linkable-channels-and-entities).|
+
+Linking allows users to continue conversations on their preferred channels. An appUser's linked channels will be found in the `clients` field. When a link request is first made, the channel will be added to the `pendingClients` field. The appUser is then prompted to accept the linking request. If they do so, the corresponding channel is then moved from the `pendingClients` field to `clients` field.
```
Comment (mspensieri) [inline source/rest/index.md]: What about the case where you skip verification?

File: source/rest/index.md [hunk 2]
```diff
@@ -963,6 +963,128 @@ Suppose for example you begin a conversation with an end user `bob@example.com`
...
+
+```
+200 OK
+```
+
+<api>`DELETE /v1/appusers/{smoochId|userId}/channels/{channel}`</api>
+
+Removes the specified channel from the appUser's clients.
+
+## Ping an App User's Linked Channel
```
Comment (mspensieri) [inline source/rest/index.md]: Don't think this needs to be documented, it's a very narrow use case and you can't even customize the text

File: source/rest/index.md [hunk 3]
```diff
@@ -963,6 +963,128 @@ Suppose for example you begin a conversation with an end user `bob@example.com`
...
+
+> Request:
+
+```shell
+curl https://api.smooch.io/v1/appusers/deb920657bbc3adc3fec7963/channels/twilio \
```
Comment (alavers) [inline source/rest/index.md]: Agree with Mike's comment above. I'm also not sure this API route is correct

**Comment by alavers** [comment]
Rest looks 👍

File: source/rest/index.md (hunk 2 above)
Comment (jpjoyal) [inline source/rest/index.md]: Yes I would not document this

File: source/rest/index.md [hunk 4]
```diff
@@ -963,6 +963,128 @@ Suppose for example you begin a conversation with an end user `bob@example.com`
...
+
+Linking allows users to continue conversations on their preferred channels. An appUser's linked channels will be found in the `clients` field. When a link request is first made, the channel will be added to the `pendingClients` field. The appUser is then prompted to accept the linking request. If they do so, the corresponding channel is then moved from the `pendingClients` field to `clients` field.
+
+### Linkable channels and entities
+
+As of now, appUsers can only be linked to the twilio channel.
+
+| Channel type                 | Required entity              |
+|------------------------------|------------------------------|
+| twilio                       | phoneNumber                  |
```
Comment (jpjoyal) [inline source/rest/index.md]: I would add details about the phone number format required. (what validation is the api doing?) Maybe we can piggy back on twillo's doc too

File: source/rest/index.md [hunk 5]
```diff
@@ -963,6 +963,107 @@ Suppose for example you begin a conversation with an end user `bob@example.com`
...
+|-------------------------------|----------------------------|
+| **type**<br/><span class='req'>required</span>     | The channel to link.|
+| **{entity}**<br/><span class='req'>required</span> | The required entity for linking. This is [different for each channel](#linkable-channels-and-entities).|
+| **{skipConfirmation}**<br/><span class='opt'>optional</span> | Flag to specify whether or not the confirmation message is sent. Requires app scope JWT.|
+
+Linking allows users to continue conversations on their preferred channels. An appUser's linked channels will be found in the `clients` field. 
+
+When a link request is first made, the channel will be added to the `pendingClients` field. The appUser is then prompted to accept the linking request. If they do so, the corresponding channel is then moved from the `pendingClients` field to `clients` field. If they reject the linking request then the channel is removed from `pendingClients`.
+
+It is posible to skip the confirmation step with the `skipConfirmation` flag. App scope JWT is required to specify the confirmation option.
```
Comment (mspensieri) [inline source/rest/index.md]: `posible`

**Comment by mspensieri** [comment]
👍

File: source/rest/index.md [hunk 6]
```diff
@@ -963,6 +963,107 @@ Suppose for example you begin a conversation with an end user `bob@example.com`
...
+}
+```
+
+<api>`POST /v1/appusers/{smoochId|userId}/channels`</api>
+
+| **Arguments**                 |                            |
+|-------------------------------|----------------------------|
+| **type**<br/><span class='req'>required</span>     | The channel to link.|
+| **{entity}**<br/><span class='req'>required</span> | The required entity for linking. This is [different for each channel](#linkable-channels-and-entities).|
+| **{skipConfirmation}**<br/><span class='opt'>optional</span> | Flag to specify whether or not the confirmation message is sent. Requires app scope JWT.|
```
Comment (alavers) [inline source/rest/index.md]: should be `skipConfirmation` (without curly braces) no?

**Comment by chloepouprom** [comment]
Added some info on phone number validation @jpjoyal https://github.com/smooch/smooch-docs/pull/116/files#diff-e68daa59411c75d0bd3a4e0bfd3e904dR1036

**Comment by alavers** [comment]
👍

**Comment by jugarrit** [comment]
Got the 👍  from @jpjoyal 
And me 👍

**Pull request merged**